src/d3a/models/area/__init__.py

```diff
@@ -128,7 +127,8 @@ def activate(self, bc=None):
             self.log.debug("No strategy. Using inter area agent.")
         self.log.debug('Activating area')
         self.active = True
-        self.dispatcher.broadcast_activate()
+        if len(self.children) != 0:
```

Comment:
Is this still needed ? I think that broadcast_activate deals already with areas that do not have children.

Reply:
Not really, but it saves two function calls for each leaf area. Do you think this is worth it to keep it? In case of the tick event this could make a difference in runtime, right?

Reply:
Not really, 2 function calls should have really negligible performance impact here. It is better IMO to go for readability in this one, and to not distract the reader of the method by additional checks.